superdesk/editor_utils.py

```diff
@@ -401,29 +465,170 @@ def __init__(self, item, field='body_html'):
+                        create_atomic_block('EMBED', {'data': {'html': html}})
+                        continue
+                    else:
+                        logger.warning('ignore block %s', str(elem.tag))
+                        continue
+                block_text = elem.text or ""
+                inline_style_ranges = []
+                entity_ranges = []
+                for child in elem:
+                    child_text = "".join(child.itertext())
+
+                    if child.tag in TAG_STYLE_MAP:
+                        inline_style_ranges.append({
+                            'offset': len(block_text),
+                            'length': len(child_text),
+                            'style': TAG_STYLE_MAP[child.tag],
+                        })
+
+                    if child.tag in TAG_ENTITY_MAP:
+                        if not child_text:
+                            child_text = " "  # must be non-empty
+                        if child.tag == 'a':
+                            data = {'link': {'href': child.attrib.get('href'), 'target': child.attrib.get('target')}}
+                        else:
+                            data = {}
+                        entity_key = create_entity(TAG_ENTITY_MAP[child.tag], data)
+                        entity_ranges.append({'offset': len(block_text), 'length': len(child_text), 'key': entity_key})
+
+                    if child.tag == 'li':
+                        child_type = BLOCK_TYPES.UNORDERED_LIST_ITEM if elem.tag == 'ul' \
+                            else BLOCK_TYPES.ORDERED_LIST_ITEM
+                        block = self.create_block(child_type, text=child_text).data
+                        block.update(depth=depth)
+                        self.content_state['blocks'].append(block)
+
+                    block_text += child_text
+
+                    if child.tail:
+                        block_text += child.tail
+
+                if elem.tail:
+                    block_text += elem.tail
+
+                if block_type:  # no block type for ul/ol
+                    block = self.create_block(block_type, text=block_text).data
+                    block['inlineStyleRanges'] = inline_style_ranges
+                    block['entityRanges'] = entity_ranges
+                    self.content_state['blocks'].append(block)
+        else:
+            for line in value.split('\n'):
+                self.content_state['blocks'].append(self.create_block(BLOCK_TYPES.UNSTYLED, text=line).data)
+
     @property
     def html(self):
         return self.html_exporter.render()
 
+    @property
+    def text(self):
+        return '\n'.join([block.text for block in self.blocks])
+
     def get_next_entity_key(self):
         """Return a non existing key for entityMap"""
         return max((int(k) for k in self.content_state['entityMap'].keys()), default=-1) + 1
 
-    def update_item(self):
-        self.item[self.field] = self.html
+    def update_item(self, html=True):
```

Comment:
why do we have to specify again HTML when we already did in `__init__`? We may want to convert a text field to HTML or vice versa, but in this case the `self.field` should change too, right?

Reply:
my bad, didn't have it in init at first, only added it later